origin-docs/source/includes/resources/marketplace.md

```diff
@@ -0,0 +1,546 @@
+When a listing is successfully created, the `createListing` method takes a callback with two arguments:
+
+`confirmationCount` - the number of successfully created listings
+
+`transactionReceipt` - an object with ipfs information about the newly created listing.
+
+> Example: createListing
+
+```javascript
+
+const listing = {
+  category: "schema.forSale",
+  commission: { amount: "0", currency: "OGN" },
+  description: "ojwoifj weofijfoijfewoi qfoiqejqoidjq oidwjqdo",
+  language: "en-US",
+  listingType: "unit",
+  media: [{…}],
+  price: { amount: "0.01", currency: "ETH" },
+  schemaId: "http://schema.originprotocol.com/listing_v1.0.0",
+  subCategory: "schema.forSale.appliances",
+  title: "I am a great appliance listing",
+  unitsTotal: 1
+}
+
+const callback = (confirmationCount, transactionReceipt) => {
+  //manage response
+}
+
+> origin.marketplace.createListing({ listing }, callback)
+```
+
+### Arguments:
+
+|Name|Type|Required|Description|
+|----|-----|-----|-----|
+|**listing** |object|required|
+|**callback** | function |optional|provides args `confirmationCount` and `transactionReceipt`|
+
+## withdrawListing
+
+Withdrawing a transaction will set the `unitsAvailable` to zero. This will stop any further purchases of that Listing.
+When a listing is successfully created, the `withdrawListing` method takes a callback with two arguments:
+
+`confirmationCount` - the number of successfully created listings
+
+`transactionReceipt` - an object with ipfs information about the newly created listing.
+
+> Example: withdrawListing
+
+```javascript
+
+const listingId = "927-832"
+const data = {}
+const callback = (confirmationCount, transactionReceipt) => {
+  //manage response
+}
+
+> origin.marketplace.withdrawListing(listingId, data, callback)
+```
+
+### Arguments:
+
+|Name|Type|Required|Description|
+|----|-----|-----|-----|
+|**listingId** |string|required|id of the listing to be withdrawn|
+|**data**|object|optional|default to `{}` if not needed|
+|**callback** | function |optional|provides args `confirmationCount` and `transactionReceipt`|
+
+## getListingReviews
+
+Reviews are created by the buyer in the receipt stage and by the seller in the payout stage of the purchase process. A review consists of a required 1-5 rating and an optional reviewText text field.
+
+> Example: withdrawListing
+
+```javascript
+
+const listingId = "927-832"
+
+> origin.marketplace.getListingReviews(listingId)
+
+//returns
+
+[{
+  "schemaId": "http://schema.originprotocol.com/review_v1.0.0",
+  "rating": 4,
+  "text": "Solid Listing"
+  "reviewer": "0x29884972398479234792"
+]}
+```
+
+### Arguments:
+
+|Name|Type|Required|Description|
+|----|-----|-----|-----|
+|**listingId** |string|required|id of the listing|
+
+## getNotifications
+
+Each Notification corresponds to a state change of a Purchase. Notifications are currently generated for each of the following purchase stages:
+
+- seller_listing_purchased
```

Comment:
These events now look something like [this](https://github.com/OriginProtocol/origin/blob/7de3e3f336b2d12a1a58314ea0b303e2133a5559/origin-js/contracts/contracts/marketplace/v00/Marketplace.sol#L27-L39).